versioned/interface.ts

```diff
@@ -125,10 +138,23 @@ abstract class Versioned {
             ),
           }))
         );
-    } else {
+    } else if (c.version === "0.3.2") {
       return fetch(
-        `${TZKT_API_URL}/v1/bigmaps/${bigmapId}/keys?value.state.in=[{"executed":{}}, {"rejected":{}}, {"expired": {}}]${common}`
-      ).then(res => res.json());
+        `${TZKT_API_URL}/v1/contracts/events?contract=${address}&tag=proof_of_event${common}`
+      )
+        .then(res => res.json())
+        .then((events: Array<proofOfEvent>) =>
+          events.map(event => ({
+            key: event.payload.challenge_id,
+            value: proposalSchema_0_3_2.Execute(
+              unpackDataBytes({
+                bytes: event.payload.payload,
+              })
+            ),
+          }))
+        );
+    } else {
```

Comment:
I think we can remove this and do a condition on the schema: `(c.version === '0.3.1' ? proposalSchema_0_3_1 : proposalSchema_0_3_2)`

Reply:
I prefer to split them as they are different cases. because:
- otherwise, next time if there is new version, the conditions will be complicated.  
- also you will have to check condition multiple times.

In the future,  I think we refactor "proposalType" to solve this kind of thing.